src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -103,7 +103,7 @@ public VSSolutionManager()
             }
 
             UserAgent.SetUserAgentString(
-                new UserAgentStringBuilder().WithVisualStudioSKU(VSVersionHelper.GetFullVsVersionString()));
+                new UserAgentStringBuilder().WithVisualStudioSKU(VSVersionHelper.GetFullVsVersionString(_dte)));
```

Comment:
We shouldn't perform any DTE queries in a constructor which is hard to control in MEF. I'd suggest moving this statement to `NuGetPackage.Initialize`.